File: source/frontend/src/lib/dto/response/aggregate/ArchitecturalKnowledgeResponseDto.ts
import {ArchitecturalKnowledge} from "$lib/domain/aggregate/ArchitecturalKnowledge.svelte.js";
import {fromIssueListResponseDto, type IssueResponseDto} from "$lib/dto/response/entity/IssueResponseDto";

export function fromArchitecturalKnowledgeResponseDto(architecturalKnowledgeResponseDto : IssueResponseDto[]) : ArchitecturalKnowledge {
    const architecturalKnowledge : ArchitecturalKnowledge = ArchitecturalKnowledge.create();

    architecturalKnowledge.issueList = fromIssueListResponseDto(architecturalKnowledgeResponseDto);

    return architecturalKnowledge;
}

File: source/backend/src/main/java/ch/unisg/backend/controller/http/dto/response/ArchitecturalKnowledgeResponseDto.java
package ch.unisg.backend.controller.http.dto.response;

import ch.unisg.backend.controller.http.dto.response.ad.IssueResponseDto;
import ch.unisg.backend.controller.http.dto.response.ar.ArchitectureRequirementResponseDto;
import ch.unisg.backend.core.domain.aggregate.ArchitecturalKnowledge;

import java.util.HashMap;

public class ArchitecturalKnowledgeResponseDto extends ResponseDto {

    public static HashMap<String, Object> toJson(ArchitecturalKnowledge architecturalKnowledge) {

        HashMap<String, Object> result = new HashMap<>();

        result.put("issueList", IssueResponseDto.toJson(architecturalKnowledge.getIssueList()));
        result.put("architectureRequirementList", ArchitectureRequirementResponseDto.toJson(architecturalKnowledge.getArchitectureRequirement()));

        return result;

    }
}
